internal/pkg/util/util.go

```diff
@@ -15,3 +16,19 @@ func Accept(l plugin_v1.Listener) (net.Conn, error) {
 	}
 	return conn, err
 }
+
+// Returns a function that will noop when debugEnabled=false, and will log
+// the given msg using `log.Print` or `log.Printf` (depending on the number
+// arguments given) debugEnabled=true
+func OptionalDebug(debugEnabled bool) func(string, ...interface{}) {
+	return func(msg string, args ...interface{}) {
+		if !debugEnabled {
```

Comment:
Can we move this conditional into the parent function?  And conditionally return a noop function. Otherwise LGTM!

Reply:
Excellent point!